### ISSUE TITLE
GPL license on fy-list.h

### ISSUE THREAD

**Issue by Enjection** [pantoniou/libfyaml#63]
It seem's that it is forbidden to licesne your library under MIT license with code licensed under GPL2.0 in it.

**Comment by pantoniou** [comment]
Up to now people didn't object much (GPL2 is much less viral than GPL3).

Another reason is that using the Linux kernel's list implementation makes things easier to integrate in the kernel (when/if that time comes).

For commercial applications a pure MIT library has merits.

I will continue this discussion at the pull request...

**Comment by bkmgit** [comment]
Maybe it is helpful to indicate in the README that fy-list.h is GPL2 and and [xxhash](https://github.com/pantoniou/libfyaml/blob/master/src/xxhash/xxhash.c) is BSD-2-Clause

**Comment by zlobober** [comment]
I agree with @Enjection, @bkmgit and the most commenters from PR #64 that it is not nice to keep incorrect licensing information. This could possibly poison third-party projects with permissive licenses that depend on libfyaml, making them implicitly violate GPL license rules, as they rely on you [stating](https://github.com/pantoniou/libfyaml/blob/master/LICENSE) that your project is MIT-licensed. 

To me this issue should be qualified as critical for an open-source project. I suggest merging either the PR #89, switching the library license to GPL or (which is better) PR #64, getting rid of GPL code and enabling easy usage of a library in open-source projects with permissive licenses.
